subprojects/core-api/src/main/java/org/gradle/process/ConfigurableJavaForkOptions.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.tasks.Nested;
+
+import java.util.List;
+
+/**
+ * Java fork options which are exposed to the user for configuration.
+ *
+ * @since 4.6
+ */
+@Incubating
+public interface ConfigurableJavaForkOptions extends JavaForkOptions {
+
+    /**
+     * Command line argument providers for the java process to fork.
+     */
+    @Nested
+    List<CommandLineArgumentProvider> getJvmArgProviders();
```

Comment:
I think I'd prefer if we used `argument` vs `arg`. WDYT?

Reply:
Changed to `argument`. I used `arg` since we have `jvmArgs` in `JavaForkOptions`.